src/Land/Model/Runoff.jl

```diff
@@ -0,0 +1,91 @@
+module Runoff
+
+using ...VariableTemplates
+using DocStringExtensions
+
+export AbstractPrecipDistribution,
+    ConstantPrecip, AbstractSurfaceRunoffModel, NoRunoff, compute_surface_flux
+
+"""
+    AbstractPrecipDistribution{FT <: AbstractFloat}
+"""
+abstract type AbstractPrecipDistribution{FT <: AbstractFloat} end
+
+"""
+    ConstantPrecip{FT, F} <: AbstractPrecipDistribution{FT}
+
+Instance of a precipitation distribution where the precipication value
+is constant across the domain. However, this value can change in time.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct ConstantPrecip{FT, F} <: AbstractPrecipDistribution{FT}
+    "Mean precipitation in grid"
+    mp::F
+    function ConstantPrecip{FT}(; mp::Function = (t) -> FT(0)) where {FT}
+        f = (t) -> FT(mp(t))
+        new{FT, typeof(f)}(f)
+    end
+end
```

Comment:
it might be a bit cleaner to define this as a callable object (so that you can enforce some properties, like the constant precip is actually a constant) and provide some encapsulation.

``` 
struct ConstantPrecip{FT} <:  AbstractPrecipDistribution{FT}
    v::FT
end

function (p::ConstantPrecip{FT})(time::FT) where {FT}
    return p.v * time
end
```

Reply:
ok! im doing something very similar now:
```
struct DrivenConstantPrecip{FT, F} <: AbstractPrecipModel{FT}
    "Mean precipitation in grid"
    mp::F
    function DrivenConstantPrecip{FT}(mp::F) where {FT,F}
        new{FT,F}(mp)
    end
end

function (dcp::DrivenConstantPrecip{FT})(t::Real) where {FT}
    return FT(dcp.mp(t))
end
```

Reply:
Then it is called as precip_model(t), instead of precip_model.mp(t).

thanks!